wagtail/documents/wagtail_hooks.py

```diff
@@ -190,7 +190,7 @@ def check_view_restrictions(document, request):
 
                 password_required_template = getattr(
                     settings,
-                    "DOCUMENT_PASSWORD_REQUIRED_TEMPLATE",
+                    "WAGTAIL_DOCUMENT_PASSWORD_REQUIRED_TEMPLATE",
```

Comment:
We will need to have a backwards compatible lookup of the old value here. 

And if the old value is in use we need to raise a depreciation warning.